tests/console/yast2_cmdline.pm

```diff
@@ -40,8 +42,25 @@ sub run_yast_cli_test {
 }
 
 sub run {
+    my $cmd;
+
     select_console 'root-console';
 
+    if (is_sle) {
+        if (is_sle('15+') and get_var('REPO_SLE15_MODULE_BASESYSTEM_SOURCE')) {
+            $cmd = "ar -f " . "$utils::OPENQA_FTP_URL/" . get_var('REPO_SLE15_MODULE_BASESYSTEM_SOURCE') . " repo-source";
+        }
+        elsif (is_sle('<15') and get_var('REPO_SLES_SOURCE')) {
+            $cmd = "ar -f " . "$utils::OPENQA_FTP_URL/" . get_var('REPO_SLES_SOURCE') . " repo-source";
+        }
```

Comment:
So you did not like my proposal

```perl
if (is_sle) {
    zypper_call('ar -f ' . $utils::OPENQA_FTP_URL/ . get_var('REPO_SLE15_MODULE_BASESYSTEM_SOURCE', get_required_var('REPO_SLES_SOURCE')) . ' repo-source');
}
```

?

Reply:
It doesn't work
http://dhcp151.suse.cz/tests/3503#step/yast2_cmdline/3

Reply:
I see. Of course the get_required_var is executed in any case. Does the following work?:

```perl
my $repo = get_var('REPO_SLE15_MODULE_BASESYSTEM_SOURCE');
$repo //= get_required_var('REPO_SLES_SOURCE');
zypper_call('ar -f ' … $repo …);
```

If not how about:

```perl
my $repo = get_var('REPO_SLE15_MODULE_BASESYSTEM_SOURCE');
$repo //= get_var('REPO_SLES_SOURCE', '');
die "Need valid source repo in either REPO_SLE15_MODULE_BASESYSTEM_SOURCE or REPO_SLES_SOURCE" unless $repo;
zypper_call('ar -f ' … $repo …);
```

Reply:
I have almost missed this one, let me try it

Reply:
It works fine http://dhcp151.suse.cz/tests/3516. With respect to the complexity of logic, I would leave it as it is for now. We can create another ticket to refactor this logic if it is necessary.

Reply:
Please not get into the discussion about tickets for refactoring. We can live with it concerning this PR but then we should work on this before we can close the related *ticket*